loopy/target/c/__init__.py

```diff
@@ -1264,7 +1304,19 @@ class CTarget(CFamilyTarget):
     """This target may emit code using all features of C99.
     For a target base supporting "least-common-denominator" C,
     see :class:`CFamilyTarget`.
+
+    .. attribute:: has_gnu_libc
+
+        An instance of :class:`bool` indicating whether the kernel will be
+        compiled with C-compiler that's built with GNU-libc as its standard
+        C-library.
     """
+    hash_fields = CFamilyTarget.hash_fields + ("has_gnu_libc",)
+    comparison_fields = CFamilyTarget.comparison_fields + ("has_gnu_libc",)
+
+    def __init__(self, fortran_abi=False, has_gnu_libc=False):
+        super().__init__(fortran_abi=fortran_abi)
+        self.has_gnu_libc = has_gnu_libc
```

Comment:
Similar to the ExecutableCTarget I wanted to add a mechanism here that guesses if we have gnu libc. So something like this
```
        from loopy.target.c.c_execution import CCompiler
        self.compiler = CCompiler()
        if has_gnu_libc is None:
            has_gnu_libc = self.compiler.toolchain.cc in ["gcc", "g++"]
```
This doesn't work it said `Default toolchain guessed from python config not found, replacing with default GCCToolchain.`

Is there a way to get this to work? Or is there a reason why this should not work?

The reason that I wanted that is that in Firedrake we generate everything with `CTargets`. If this was able to guess on it's own that gnulibc is available, less changes would be required on Firedrake side.

Reply:
Tried doing a more automatic job of inferring `glibc` support in Loopy, but that was coming with some non-trivial assumptions. However, PyOP2 seems to already have a well-defined [set of rules](https://github.com/OP2/PyOP2/blob/13f0d336fa6ff85c93196c9dc50e12406e46b526/pyop2/compilation.py#L484-L499) to infer the C-compiler, using that would be easier I think.

An option could have been to default `has_gnu_libc` to True, but I think that would emit incorrect code according to the C-standard.

Reply:
Okay! Then I will hook this up in PyOP2 by going through `CTarget` with `gnulibc=True` if our compiler sniff shows that we use gcc or g++.